java/src/jmri/jmrix/sprog/SprogProgrammer.java

```diff
@@ -259,8 +255,17 @@ synchronized public void notifyVersion(SprogVersion v) {
         _sv = v;
         // Save it for others
         _memo.setSprogVersion(v);
-        if (log.isDebugEnabled()) {
-            log.debug("Found: {}", v.toString());
+        log.debug("Found: {}", v.toString());
+
+        if (!_sv.supportsCVHints()) {
+            log.debug("Hardware does not support hints");
+           _startVal = 0;
+        }
+        try {
+            readCVWithDefault(_cv, _progListener, _startVal);
+        }
+        catch (jmri.ProgrammerException e) {
+            throw new IllegalArgumentException("Unthrown Programmer Exception " + e.getMessage());
```

Comment:
Should it be `Unknown` and not `Unthrown`?

Reply:
Sorry, I misread the string. But I suggest you change this to:
`throw new IllegalArgumentException("Programmer Exception " + e.getMessage(), e);`
if you keep it.

Adding e as a parameter means that the stack trace will include the original exception as the cause.